python/lsst/pipe/tasks/characterizeImage.py

```diff
@@ -0,0 +1,493 @@
+    )
+    doWriteExposure = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Write icExp and icExpBackground in addition to icSrc? Ignored if doWrite False.",
+    )
+    psfIterations = pexConfig.RangeField(
+        dtype = int,
+        default = 2,
+        min = 1,
+        doc = "Number of iterations of detect sources, measure sources, estimate PSF. "
+            "If useSimplePsf='all_iter' then 2 should be plenty; otherwise more may be wanted.",
+    )
+    background = pexConfig.ConfigField(
+        dtype = estimateBackground.ConfigClass,
+        doc = "Configuration for initial background estimation",
+    )
+    detectAndMeasure = pexConfig.ConfigurableField(
+        target = DetectAndMeasureTask,
+        doc = "Detect and measure sources, for measuring PSF",
+    )
+    useSimplePsf = pexConfig.Field(
+        dtype = bool,
+        default = True,
+        doc = "Replace the existing PSF model with a simplified version that has the same sigma "
+            "at the start of each PSF determination iteration? Doing so makes PSF determination "
+            "converge more robustly and quickly.",
+    )
+    installSimplePsf = pexConfig.ConfigurableField(
+        target = InstallGaussianPsfTask,
+        doc = "Install a simple PSF model",
+    )
+    astrometry = pexConfig.ConfigurableField(
+        target = AstrometryTask,
+        doc = "Task to load and match reference objects. Only used if measurePsf can use matches. "
+            "Warning: matching will only work well if the initial WCS is accurate enough "
+            "to give good matches (roughly: good to 3 arcsec across the CCD).",
+    )
+    measurePsf = pexConfig.ConfigurableField(
+        target = MeasurePsfTask,
+        doc = "Measure PSF",
+    )
+    repair = pexConfig.ConfigurableField(
+        target = RepairTask,
+        doc = "Remove cosmic rays",
+    )
+
+    def setDefaults(self):
+        pexConfig.Config.setDefaults(self)
+        # just detect bright stars; includeThresholdMultipler=10 seems large,
+        # but these are the values we have been using
+        self.detectAndMeasure.detection.thresholdValue = 5.0
+        self.detectAndMeasure.detection.includeThresholdMultiplier = 10.0
+        # do not deblend, as it makes a mess
+        self.detectAndMeasure.doDeblend = False
+        # measure and apply aperture correction; note: measuring and applying aperture
+        # correction are disabled until the final measurement, after PSF is measured
+        self.detectAndMeasure.doMeasureApCorr = True
+        self.detectAndMeasure.measurement.doApplyApCorr = "yes"
+        # minimal set of measurements needed to determine PSF
+        self.detectAndMeasure.measurement.plugins.names = [
+            "base_PixelFlags",
+            "base_SdssCentroid",
+            "base_SdssShape",
+            "base_GaussianFlux",
+            "base_PsfFlux",
+            "base_CircularApertureFlux",
+        ]
```

Comment:
Why limit the set of plugins?  All plugins need to have aperture corrections measured.  If they're not measured here, then they can't be corrected later.  So I think you need to be far more inclusive, at least for the fluxes.